dockstore-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java

```diff
@@ -291,6 +292,42 @@ public void testCWLLanguageParsingInformation() {
         Assert.assertTrue(parsedInformation.isHasHTTPImports());
     }
 
+    /**
+     * Test the precedence of CWL requirements and hints.
+     * The workflow steps in the referenced workflow are named to indicate whether the step and invoked command
+     * defined requirements or hints.  For example, the workflow step id "requirement_none" means that the workflow
+     * step defined a requirement, and the invoked command did not define a requirement or a hint.
+     *
+     * <p> According to the CWL spec, a child inherits requirements and hints from its parent.  Child requirements
+     * override parent requirements, child hints override parent hints, and if both are present at a node,
+     * requirements take precedence over hints.
+     */
+    @Test
+    public void testCWLRequirementsAndHints() {
+        final ApiClient webClient = getWebClient(USER_2_USERNAME, testingPostgres);
+        WorkflowsApi workflowApi = new WorkflowsApi(webClient);
+        Workflow cwlWorkflow = workflowApi
+               .manualRegister(SourceControl.GITHUB.name(), "dockstore-testing/cwl-requirements-and-hints", "/parent.cwl", "CWL",
+                       DescriptorLanguage.CWL.toString(), "/parent.json");
+        Long cwlId = cwlWorkflow.getId();
+        workflowApi.refresh(cwlId, false);
+        Workflow workflow = workflowApi.getWorkflow(cwlId, null);
+        WorkflowVersion version = workflow.getWorkflowVersions().stream().findFirst().get();
```

Comment:
what is the "first" version? might this break in the future if someone adds a new branch to `dockstore-testing/cwl-requirements-and-hints"`

Reply:
True that, will fix.